### ISSUE TITLE
Once per hour playlists only play on the first hour

### ISSUE THREAD

**Issue by dvdokkum** [AzuraCast/AzuraCast#2536]
Just want to say thank you so much for this amazing software! I'm setting it up for WXYC ([the first radio station in the world to broadcast online](https://twitter.com/DVD/status/1192449673572888576)) to use as a backup remote broadcasting solution during these COVID-19 times when our main studio is less accessible to our student DJs. Please let me know if I can provide any more details on this bug.

**Using Docker installation method**
Yes

**Host Operating System**
Ubuntu 18.04.3

**Describe the bug**
Once per hour playlists only play on the first hour after restarting liquidsoap with the new config. It does not play on any subsequent hours. This is on a fresh DigitalOcean install that was updated to 0.9.9.

**To Reproduce**

1. Create a playlist with type "once per hour"
2. Set minute of hour to play = 0 (also doesn't work for other minutes)
3. See screenshots below for all playlist settings...
4. Confirm that it plays at the right time on the first hour after restarting the AutoDJ Service
5. Wait an hour, notice it doesn't play at all on the appointed minute.

**Expected behavior**
Expected behavior is for one of the 6 songs in the linked media folder to play at random every hour, on the hour.

**Relevant Logs**
Here's the first play after I restarted the AutoDJ, sounds good:

```2020/03/19 17:59:51 [lang:3] AzuraCast Raw Response: annotate:title="Coat Of Stories",artist="Lula Côrtes",duration="132.",song_id="e49e830d5dfdb6e0fca5f2d3d729d003",media_id="57",playlist_id="4":/var/azuracast/stations/main/media/freeform/03_coat_of_stories.mp3
2020/03/19 17:59:51 [decoder:3] Method "MAD" accepted "/var/azuracast/stations/main/media/freeform/03_coat_of_stories.mp3".
2020/03/19 18:00:00 [playlist_hourly_station_id:3] Prepared "/var/azuracast/stations/main/media/station IDs/46_-_bill_burton_-_track46.wav" (RID 3).
2020/03/19 18:00:00 [main_interrupt_switch:3] Switch to sequence_7705 with transition.
2020/03/19 18:00:00 [decoder:3] Method "WAV" accepted "/var/azuracast/stations/main/media/station IDs/34_-_gene_roddenbury_-_track34.wav".
2020/03/19 18:00:18 [cue_playlist_hourly_station_id:3] Cueing out...
2020/03/19 18:00:18 [main_interrupt_switch:3] Switch to main_autodj_fallback with forgetful transition.
2020/03/19 18:00:49 [main_next_song:3] Prepared "/var/azuracast/stations/main/media/freeform/03_coat_of_stories.mp3" (RID 2).
2020/03/19 18:00:49 [lang:3] AzuraCast Feedback Response: OK
```
And then here's the next hour, nothing:
```
2020/03/19 18:54:09 [main_next_song:3] Prepared "/var/azuracast/stations/main/media/freeform/03_relativity_suite_part_1.mp3" (RID 3).
2020/03/19 18:54:09 [lang:3] AzuraCast Feedback Response: OK
2020/03/19 19:00:22 [lang:3] AzuraCast Raw Response: annotate:title="Love",artist="The Sundays",duration="272.",song_id="ab065a706064bcad7fe2f87b09f3f9c3",media_id="79",playlist_id="4":/var/azuracast/stations/main/media/freeform/the_sundays_-_love.mp3
2020/03/19 19:00:22 [decoder:3] Method "MAD" accepted "/var/azuracast/stations/main/media/freeform/the_sundays_-_love.mp3".
2020/03/19 19:01:02 [main_next_song:3] Prepared "/var/azuracast/stations/main/media/freeform/the_sundays_-_love.mp3" (RID 1).
2020/03/19 19:01:02 [lang:3] AzuraCast Feedback Response: OK
```

Here's what I think is the relevant part of the Liquid Soap config file:

```
playlist_hourly_station_id = playlist(id="playlist_hourly_station_id",mode="random",reload_mode="watch",conservative=true,default_duration=10.,length=20.,"/var/azuracast/stations/main/playlists/playlist_hourly_station_id.m3u")
playlist_hourly_station_id = audio_to_stereo(id="stereo_playlist_hourly_station_id", playlist_hourly_station_id)
playlist_hourly_station_id = cue_cut(id="cue_playlist_hourly_station_id", playlist_hourly_station_id)
playlist_hourly_station_id = drop_metadata(playlist_hourly_station_id)

# Standard Playlists
radio = random(id="main_standard_playlists", weights=[3], [playlist_freeform])

# AutoDJ Next Song Script
def azuracast_next_song() =
    uri = list.hd(get_process_lines("curl -s --request POST --url http://web/api/internal/1/nextsong --form api_auth="^string.quote("XXXXXXXXXXXXXXXXXXXXXXXXXXXXX")^""), default="")
    log("AzuraCast Raw Response: #{uri}")
    
    if uri == "" or string.match(pattern="Error", uri) then
        log("AzuraCast Error: Delaying subsequent requests...")
        system("sleep 2")
        request.create("")
    else
        request.create(uri)
    end
end

dynamic = request.dynamic(id="main_next_song", timeout=20., azuracast_next_song)
dynamic = audio_to_stereo(id="main_stereo_next_song", dynamic)
dynamic = cue_cut(id="main_cue_next_song", dynamic)
radio = fallback(id="main_autodj_fallback", track_sensitive = true, [dynamic, radio])

# Interrupting Schedule Switches
radio = switch(id="main_interrupt_switch", track_sensitive=false, [ ({ 0m }, once(playlist_hourly_station_id)), ({true}, radio) ])

requests = request.queue(id="main_requests")
requests = audio_to_stereo(id="main_stereo_requests", requests)
requests = cue_cut(id="main_cue_requests", requests)
radio = fallback(id="main_requests_fallback", track_sensitive = true, [requests, radio])

add_skip_command(radio)

radio = fallback(id="main_safe_fallback", track_sensitive = false, [radio, single(id="error_jingle", "/usr/local/share/icecast/web/error.mp3")])
```

**Screenshots**

*All Playlists*
<img width="896" alt="Screen Shot 2020-03-19 at 9 08 06 PM" src="https://user-images.githubusercontent.com/1800276/77128587-0944e400-6a27-11ea-8b56-08ed8510b7e4.png">

*Hourly Playlist Settings*
<img width="694" alt="Screen Shot 2020-03-19 at 9 07 33 PM" src="https://user-images.githubusercontent.com/1800276/77128606-15c93c80-6a27-11ea-9c89-f45293879b73.png">
<img width="688" alt="Screen Shot 2020-03-19 at 9 07 46 PM" src="https://user-images.githubusercontent.com/1800276/77128613-1a8df080-6a27-11ea-845c-72254e03605f.png">

**Additional context**
Again, really appreciate the help with this one. I'm pretty new to Azuracast, but have been following the project on Twitter for a while. I hope this is a real bug and not just a newbie error. FWIW, I'm getting the same issue with the latest update `#c851f92` from earlier today.

**Comment by AmsterdamFM** [comment]
We have exactly the same problem. But the half an hour news (prio 2) is played and is just a short clip but the top of the hour news (prio 25) at around :00 is never played. No other playlist has that priority.

**Comment by dvdokkum** [comment]
Still trying to figure this one out.... fwiw, I tried removing all of the songs from the hourly playlist except for one just to see if that helped at all. No dice. Same behavior even if there is only one song in the playlist. I guess that makes sense, since I don't even see the playlist logic firing in the LS logs on the hour. Will keep changing random settings until maybe it will work?

**Comment by dvdokkum** [comment]
We're working around this bug right now by explicitly scheduling times for the playlist run via the playlist schedule mode. At first I tried to do all 24 hour marks in a single playlist, but this resulted in liquidsoap errors on restarting the backend. There must be some kind of problem with having too many scheduled times for a given playlist... ultimately we got it to work by spreading out the 24 slots across 5 different playlists. Not ideal, but I guess as a work around it gets the job done of playing an audio file at the top of every hour.
